qemu/tests/cfg/slof_user_interface.cfg

```diff
@@ -18,59 +18,96 @@
-            menu_option2 = "disk pci scsi ${drive_pci_addr_stg0}"
-            menu_option3 = "disk pci scsi ${virtio_scsi_pci_addr_image1}"
-            menu_option4 = "disk vdevice v-scsi ${spapr_vscsi_reg}"
-            bootable_index = 3
+            virtio_net:
+                nic_pci_addr_nic1 = 0xa
+                menu_option5 = "net pci ethernet ${nic_pci_addr_nic1}"
+            variants:
+                - @default:
+                    only virtio_scsi
+                    no spapr_vscsi
+                    virtio_scsi_pci_addr_image1 =  0xe
+                    bus_extra_params_image1 = 'addr=${virtio_scsi_pci_addr_image1}'
+                    images += " stg0 stg1"
+                    drive_pci_addr_stg0 = 8
+                    drive_format_stg0 = virtio
+                    image_name_stg0 = images/stg0
+                    image_size_stg0 = 25G
+                    remove_image_stg0 = yes
+                    force_create_image_stg0 = yes
+                    spapr_vscsi_reg =  0x71000001
+                    scsi_hba_stg1 = spapr-vscsi
+                    bus_extra_params_stg1 = 'reg=${spapr_vscsi_reg}'
+                    image_name_stg1 = images/stg1
+                    image_size_stg1 = 25G
+                    remove_image_stg1 = yes
+                    force_create_image_stg1 = yes
+                    cdroms = 'cd1'
+                    cdrom_cd1 = /tmp/new.iso
+                    pre_command_noncritical = no
+                    pre_command = 'dd if=/dev/urandom of=/tmp/new bs=10M count=1 &&'
+                    pre_command += ' mkisofs -o /tmp/new.iso /tmp/new'
+                    post_command = "rm -rf /tmp/new.iso /tmp/new"
+                    virtio_scsi_pci_addr_cd1 =  0xb
+                    drive_bus_cd1 = 1
+                    bus_extra_params_cd1 = 'addr=${virtio_scsi_pci_addr_cd1}'
+                    spapr-vlan:
+                        spapr_vlan_reg = 0x71000003
+                        nic_extra_params_nic1 = 'reg=${spapr_vlan_reg}'
+                        menu_option5 = "net vdevice l-lan ${spapr_vlan_reg}"
+                    boot_dev_num = 5
+                    menu_option1 = "cdrom pci scsi ${virtio_scsi_pci_addr_cd1}"
+                    menu_option2 = "disk pci scsi ${drive_pci_addr_stg0}"
+                    menu_option3 = "disk pci scsi ${virtio_scsi_pci_addr_image1}"
+                    menu_option4 = "disk vdevice v-scsi ${spapr_vscsi_reg}"
+                    bootable_index = 3
+                - with_vscsi_cdrom:
+                    only spapr_vscsi
+                    only virtio_net
+                    spapr_vscsi_reg =  0x71000001
+                    bus_extra_params_image1 = 'reg=${spapr_vscsi_reg}'
+                    cdroms = 'cd1'
+                    cdrom_cd1 = /tmp/new.iso
+                    drive_bus_cd1 = 0
+                    drive_port_cd1 = 1
+                    pre_command_noncritical = no
```

Comment:
L51-L56 and L75-L76, L79-L82 are the same, can we mvoe them to upper layer?

Reply:
The reason I did not move them to upper layer is considering the expandability of "send_f12", because that maybe other cases which have different disks need to added to it later, maybe others don't have cdrom.